checkov/common/output/baseline.py

```diff
@@ -67,9 +69,15 @@ def compare_and_reduce_reports(self, scan_reports: list[Report]) -> None:
             scan_report.skipped_checks = [
                 check for check in scan_report.skipped_checks if self._is_check_in_baseline(check)
             ]
-            scan_report.failed_checks = [
-                check for check in scan_report.failed_checks if not self._is_check_in_baseline(check)
-            ]
+            failed_checks = []
+            for check in scan_report.failed_checks:
+                if not self._is_check_in_baseline(check):
+                    failed_checks.append(check)
+                elif self.output_skipped:
+                    check.check_result["suppress_comment"] = "baseline-skipped"
+                    check.check_result["result"] = CheckResult.SKIPPED
+                    scan_report.skipped_checks.append(check)
+            scan_report.failed_checks = failed_checks
```

Comment:
since this is a very special case, I don't think it is worth to remove the list comprehension and use the slower iteration, just run through the failed checks, if someone wants the output
```suggestion
            if self.output_skipped:
                for check in scan_report.failed_checks:
                    check.check_result["suppress_comment"] = "baseline-skipped"
                    check.check_result["result"] = CheckResult.SKIPPED
                    scan_report.skipped_checks.append(check)
```